pages/index.js

```diff
@@ -14,17 +14,67 @@ export const appName = "Paint by Text";
 export const appSubtitle = "Edit your photos using written instructions, with the help of an AI.";
 export const appMetaDescription = "Edit your photos using written instructions, with the help of an AI.";
 
+function TokenModal({ onTokenSet }) {
+  const [token, setToken] = useState("");
+  return (
+    <div className="fixed inset-0 z-50 flex items-center justify-center pointer-events-auto">
+      <form
+        className="w-full max-w-md mx-auto p-8 bg-white rounded-xl shadow-2xl border border-gray-200 relative animate-in fade-in"
+        onSubmit={e => {
+          e.preventDefault();
+          if (token) {
+            localStorage.setItem("replicateApiToken", token);
+            onTokenSet(token);
+          }
+        }}
+        aria-modal="true"
+        role="dialog"
+      >
+        <h2 className="text-2xl font-bold mb-2 text-center">Enter your Replicate API token</h2>
+        <p className="mb-4 text-center text-gray-600">
+          Get a free token at {" "}
+          <a
+            href="https://replicate.com/account/api-tokens?new-token-name=paint-by-text-kontext"
+            target="_blank"
+            rel="noopener noreferrer"
+            className="underline text-blue-600"
+          >
+            replicate.com/account/api-tokens
+          </a>
+        </p>
+        <input
+          className="w-full border rounded p-3 mb-4 text-lg"
+          type="text"
+          value={token}
+          onChange={e => setToken(e.target.value)}
+          placeholder="r8_..."
+          required
+          autoFocus
+        />
+        <button className="w-full bg-black text-white px-4 py-3 rounded text-lg font-semibold" type="submit">
+          Start painting
+        </button>
+      </form>
+    </div>
+  );
+}
+
 export default function Home() {
   const [events, setEvents] = useState([]);
   const [predictions, setPredictions] = useState([]);
   const [error, setError] = useState(null);
   const [isProcessing, setIsProcessing] = useState(false);
   const [seed] = useState(getRandomSeed());
   const [initialPrompt, setInitialPrompt] = useState(seed.prompt);
+  const [apiToken, setApiToken] = useState(null);
+  const [showTokenForm, setShowTokenForm] = useState(true);
```

Comment:
Could this be the same stateful value? We show the form if we don't have a token, right?